src/main_loop.rs

```diff
@@ -90,30 +77,54 @@ pub(crate) struct MainLoop {
 }
 
 impl MainLoop {
-    async fn get_session(&mut self) -> Result<Session, SessionError> {
+    async fn get_connection(&mut self) -> Result<(Spirc, impl Future<Output = ()>), Error> {
         let creds = self.credentials_provider.get_credentials().await;
 
-        let session_config = self.session_config.clone();
-        let cache = self.spotifyd_state.cache.clone();
-
-        Session::connect(session_config, creds, cache, true)
-            .await
-            .map(|(session, _creds)| session)
+        // TODO: expose is_group
```

Comment:
I don't think that for your use case `is_group` needs to be exposed, as it only changes the displayed icon as far as I'm aware of.

Reply:
I was thinking of the (probably niche, if that exists at all) use case, where `spotifyd` is used to drive multiple speakers, where one might want to have the icon reflect this case.

But dedicating an “entire” option to this indeed feels a little much. Maybe I can investigate, how different combinations of `device_type` and `is_group` interact with each other and maybe just add `speaker_group` or something similar as a variant in `device_type`.

Reply:
Hmm, yeah that might actually be a use case. As it only is a visual change I wouldn't really see it as important, but probably some would like to have that option. Thanks for the insight^^

> I can't close these threads, so feel free to close them whenever you think they are done :)